### PR TITLE
Add map method to OneAnd. Fix for #885

### PR DESCRIPTION
**Pull request by pvillega**
OneAnd was missing an instance of map. This commit tries to solve this.

### CODE REVIEW

**Comment by adelbertc** [comment]
:+1: - could you also replace the existing `map` definitions in the type class instances with the new `map`?

**Comment by pvillega** [comment]
will do :)

**Comment by pvillega** [comment]
Done! I'm not sure about the `val functorList: Functor[List]` I had to define for the `CoMonad` implicit. I couldn't see an existing alternative but there may be one, if so please let me know and I'll correct it tomorrow (UK time).

**Comment by adelbertc** [comment]
You probably added it because the type checker couldn't find a `Functor[List]` in implicit scope - you can get it from `import cats.std.list._`.

File: core/src/main/scala/cats/data/OneAnd.scala [hunk 1]
```diff
@@ -131,9 +137,10 @@ private[data] sealed trait OneAndInstances extends OneAndLowPriority2 {
 }
 
 trait OneAndLowPriority0 {
+  import cats.std.list._
```
Comment (adelbertc) [inline core/src/main/scala/cats/data/OneAnd.scala]: Nitpicking: would prefer this to be moved to the top

**Comment by adelbertc** [comment]
One minor comment, but :+1: on Travis green

**Comment by adelbertc** [comment]
Hm seems moving it to the top does [funky things](https://gitter.im/typelevel/cats?at=56c3acae5fa70a130ea500bc) - anyone know why this is the case? Either way, :+1: from me

**Comment by pvillega** [comment]
Fixed compilation issue when moving import to top, thanks to @julien-truffaut for his support (he was the one fixing it in fact, I'm just stealing the contribution :P)

File: core/src/main/scala/cats/data/OneAnd.scala [hunk 2]
```diff
@@ -104,7 +111,7 @@ private[data] sealed trait OneAndInstances extends OneAndLowPriority2 {
     }
 
   implicit def oneAndSemigroup[F[_]: MonadCombine, A]: Semigroup[OneAnd[F, A]] =
-    oneAndSemigroupK.algebra
+    oneAndSemigroupK(MonadCombine[F]).algebra
```
Comment (ceedubs) [inline core/src/main/scala/cats/data/OneAnd.scala]: Sorry, this is nitpicky, but could you please change this to `oneAndSemigroupK[F].algebra`? That will save a `MonadCombine.apply` call.

Comment (julien-truffaut) [inline core/src/main/scala/cats/data/OneAnd.scala]: for some reason the compiler get confused with `oneAndSemigroupK[F]` after adding `import cats.std.list._` on the top of the file. It says that there is an ambiguity between `MonadCombine[List]` and `MonadCombine[F]` ...

Comment (julien-truffaut) [inline core/src/main/scala/cats/data/OneAnd.scala]: sorry I misread your comment @ceedubs, you were right `oneAndSemigroupK[F]` is enough to disambiguate.

**Comment by julien-truffaut** [comment]
:+1: when the build is green

**Pull request merged**